src/Plugin.php

```diff
@@ -256,14 +271,65 @@ public function saveOptionsHook(): void
         (new SettingsTableAdapter())->save();
     }
 
+    /**
+     * @param string|null $languageCode
+     * @return string
+     */
+    public static function getTransientKey(string $languageCode = null): string
+    {
+        if ($languageCode) {
```

Comment:
would extract this to a helper function or own class for the language. in there we can cover the logic to determine which plugin is used or use the fallback.